src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreResult.cs

```diff
@@ -157,23 +172,26 @@ public virtual async Task CommitAsync(ILogger log, CancellationToken token)
 
             BuildAssetsUtils.WriteFiles(buildFilesToWrite, log);
 
+            if (result.LockFile == null || result.LockFilePath ==  null)
```

Comment:
This is adding a check that wasn't there before. 
Why?

Reply:
Having LockFile as null when we've bailed out the restore completely make sense to me and the right approach. Adding an additional flag for it, is what doesn't make sense to me.

Reply:
My problem with this is that it promotes bad practice, where you don't know if things are null and when they are null and what does it mean for them to be null.

Reply:
I'm out of arguments for this... I sincerely dont follow your concerns over bad practice for this. It's as simple as any other null check in an api to before consuming anything.

Lets talk about this offline and finalize it

Reply:
discussed offline, we're settled on this.